types/test/findAll.ts

```diff
@@ -0,0 +1,42 @@
+import { expectTypeOf } from 'expect-type'
+import { Sequelize, Op } from 'sequelize';
+import { User } from './models/User';
+
+async () => {
+    const users = await User.findAll({ raw: false });
+    expectTypeOf(users).toEqualTypeOf<User[]>()
```

Comment:
haven't seen `expectTypeOf` before. Not sure how this is supposed work, given what I know of ts (and how asserts work, since iirc block scoped type assertions don't currently work). 

```suggestion
    const users: User[] = await User.findAll({ raw: false });
```

Reply:
`expectTypeOf` actually performs stricter type comparison than Typescript in some cases, and is the general test pattern of sequelize.

The testing only happens in Typescript, which is fine since these files are type-only (never actually executed javascript)

Reply:
:shrug: I'll take your word for it.

Reply:
@AllAwesome497 `expectTypeOf` uses a <s>hack</s> great idea involving conditional types and rest parameters, it really works. By the way it was created by our Umzug maintainer, mmkal :)